public/app/core/history/richHistoryLocalStorageUtils.ts

```diff
@@ -0,0 +1,82 @@
+import { RichHistoryQuery } from '../../types';
+import { omit } from 'lodash';
+import { SortOrder } from '../utils/richHistoryTypes';
+
+/**
+ * Temporary place for local storage specific items that are still in use in richHistory.ts
+ *
+ * Should be migrated to RichHistoryLocalStorage.ts
+ */
+
+export const createRetentionPeriodBoundary = (days: number, isLastTs: boolean) => {
+  const today = new Date();
+  const date = new Date(today.setDate(today.getDate() - days));
+  /*
+   * As a retention period boundaries, we consider:
+   * - The last timestamp equals to the 24:00 of the last day of retention
+   * - The first timestamp that equals to the 00:00 of the first day of retention
+   */
+  const boundary = isLastTs ? date.setHours(24, 0, 0, 0) : date.setHours(0, 0, 0, 0);
+  return boundary;
+};
+
+export function filterQueriesByTime(queries: RichHistoryQuery[], timeFilter: [number, number]) {
+  return queries.filter(
+    (q) =>
+      q.ts < createRetentionPeriodBoundary(timeFilter[0], true) &&
+      q.ts > createRetentionPeriodBoundary(timeFilter[1], false)
+  );
+}
```

Comment:
we don't need to recalculate those for every element, do we?

```suggestion
export function filterQueriesByTime(queries: RichHistoryQuery[], timeFilter: [number, number]) {
  const filter1 = createRetentionPeriodBoundary(timeFilter[0], true); // probably the vars should have a different name
  const filter2 = createRetentionPeriodBoundary(timeFilter[1], false);
  return queries.filter(
    (q) =>
      q.ts < filter1
      q.ts > filter2
  );
}
```